### ISSUE TITLE
SIMD support

### ISSUE THREAD

**Issue by 0xAlcibiades** [paupino/rust-decimal#694]
This crate seems like a candidate for huge benefits from SIMD support where 4 wide instructions could be used for arithmetic on the scalar type this crate provides.

**Comment by richardscollin** [comment]
I agree, it's something I'd be interested implementing if there was a chance of getting it merged in. On some workloads I noticed a non-trivial amount of time was spent in Decimal parse (serde deserializaing a large vector of decimal values). Anything that can help improve that would be great.

There's a good blog post where someone did some benchmarks on this as well: https://vgatherps.github.io/2022-11-28-dec/

Implementing it using the portable simd api over the simd instrinsics api might also be a good idea so that there's no architecture specific code.

**Comment by paupino** [comment]
Closing in favour of #728 which has a more concrete discussion around SIMD integration. Thanks for the suggestion.
